src/current/advisories/a99561.md

```diff
@@ -0,0 +1,51 @@
+---
+title: Technical Advisory 99561
+advisory: A-99561
+summary: When adding a new column using the new declarative schema changer, CockroachDB retrieves a unique secondary index mistakenly, assuming it to be a primary index.
+toc: true
+affected_versions: v22.2.x, v23.1.0-v23.1.5
+advisory_date: 2023-07-07
+docs_area: releases
+---
+
+Publication date: {{ page.advisory_date | date: "%B %e, %Y" }}
+
+## Description
+
+In CockroachDB v20.2 and v21.1 (the first two versions to support [`ALTER PRIMARY KEY`](../v21.1/alter-primary-key.html)), there was a bug where in some cases the index automatically created to provide uniqueness for the existing primary key would have its `EncodingType` field marked as `PrimaryIndexEncoding`. This behavior was fixed in [#66599](https://github.com/cockroachdb/cockroach/pull/66599), but left secondary indexes with primary encoding.
```

Comment:
I think the first sentence is doing too much work. Perhaps the first thing is to explain that an index is automatically created to ensure that the primary key is unique, then summarize the bug and explain that it is fixed. Then the reader is set up to understand the rest of the TA. WDYT?

Reply:
I have added an initial sentence here, but I'm not sure if it is what you were envisioning. Feel free to adapt.